From a19ffc2727c0a75bd44128bfe942437386288a21 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Sun, 1 Apr 2012 03:53:17 +0200
Subject: [PATCH] build: move libsushi out of $libdir

Don't install our private introspected library in the system libdir,
since it's not supposed to be used outside of sushi.

https://bugzilla.gnome.org/show_bug.cgi?id=673269
---
 src/Makefile-sushi.am |    5 ++++-
 src/Makefile.am       |    6 +++---
 src/sushi.in          |   10 ++++++++--
 3 files changed, 15 insertions(+), 6 deletions(-)

Index: sushi-0.4.1/src/Makefile-sushi.am
===================================================================
--- sushi-0.4.1.orig/src/Makefile-sushi.am	2012-04-16 17:15:30.000000000 +0200
+++ sushi-0.4.1/src/Makefile-sushi.am	2012-04-26 22:16:05.195593573 +0200
@@ -53,11 +53,14 @@
 	cp xgen-betc sushi-enum-types.c && \
 	rm -f xgen-betc
 
-lib_LTLIBRARIES += libsushi-1.0.la
+pkglib_LTLIBRARIES += libsushi-1.0.la
 
 libsushi_1_0_la_LIBADD = \
     $(SUSHI_LIBS)
 
+libsushi_1_0_la_LDFLAGS = \
+    -avoid-version
+
 libsushi_1_0_la_CPPFLAGS = \
     $(sushi_cflags)
 
Index: sushi-0.4.1/src/Makefile.am
===================================================================
--- sushi-0.4.1.orig/src/Makefile.am	2012-04-16 17:15:30.000000000 +0200
+++ sushi-0.4.1/src/Makefile.am	2012-04-26 22:16:05.199593573 +0200
@@ -8,13 +8,13 @@
     -DSUSHI_TOP_SRCDIR=\"$(abs_top_srcdir)\" \
     -DSUSHI_BUILDDIR=\"$(abs_top_builddir)\"
 
-girdir = $(datadir)/gir-1.0
-typelibdir = $(libdir)/girepository-1.0
+girdir = $(pkgdatadir)/gir-1.0
+typelibdir = $(pkglibdir)/girepository-1.0
 
 BUILT_SOURCES =
 EXTRA_DIST =
 CLEANFILES =
-lib_LTLIBRARIES =
+pkglib_LTLIBRARIES =
 INTROSPECTION_GIRS =
 gir_DATA =
 typelib_DATA =
Index: sushi-0.4.1/src/sushi.in
===================================================================
--- sushi-0.4.1.orig/src/sushi.in	2012-04-16 17:15:30.000000000 +0200
+++ sushi-0.4.1/src/sushi.in	2012-04-26 22:16:05.199593573 +0200
@@ -7,9 +7,15 @@
 fi
 
 if test x"$GI_TYPELIB_PATH" = x ; then
-    export GI_TYPELIB_PATH=@libdir@/girepository-1.0
+    export GI_TYPELIB_PATH=@pkglibdir@/girepository-1.0
 else
-    export GI_TYPELIB_PATH=@libdir@/girepository-1.0:$GI_TYPELIB_PATH
+    export GI_TYPELIB_PATH=@pkglibdir@/girepository-1.0:$GI_TYPELIB_PATH
+fi
+
+if test x"$LD_LIBRARY_PATH" = x ; then
+    export LD_LIBRARY_PATH=@pkglibdir@
+else
+    export LD_LIBRARY_PATH=@pkglibdir@:$LD_LIBRARY_PATH
 fi
 
 if test x"$GJS_DEBUG_OUTPUT" = x ; then
Index: sushi-0.4.1/configure.ac
===================================================================
--- sushi-0.4.1.orig/configure.ac	2012-04-16 17:17:58.000000000 +0200
+++ sushi-0.4.1/configure.ac	2012-04-26 22:17:10.595591260 +0200
@@ -47,7 +47,7 @@
 GJS_JS_DIR=`$PKG_CONFIG --variable=datadir gjs-1.0`/gjs-1.0
 AC_SUBST(GJS_JS_DIR)
 
-CLUTTER_MIN_VERSION=1.10.0
+CLUTTER_MIN_VERSION=1.6.0
 GLIB_MIN_VERSION=2.29.14
 GTK_MIN_VERSION=3.0.0
 GJS_MIN_VERSION=0.7.7
